flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/SourceStreamTaskTest.java

```diff
@@ -577,45 +518,6 @@ public void testWaitsForSourceThreadOnCancel() throws Exception {
         harness.waitForTaskCompletion(Long.MAX_VALUE, true);
     }
 
-    @Test
-    public void testStopWithSavepointShouldNotInterruptTheSource() throws Exception {
```

Comment:
This is the only test I considered rewriting. I have not found a good way to do so though. First let me explain why it does not work with my changes.

The problem is the source needs to exit first in order for the sync savepoint to complete (actually to even be triggered). Thus the 
```
            notifyFuture.get();
            WasInterruptedTestingSource.allowExit();
```
hangs forever.

I don't think there is a good point now, where we can say: "ok, now we are confident the source will no longer be interrupted". At the same time, I don't think we loose much without the test. The corresponding code path is quite simple. We simply call `StreamSource#stop` which we expect will shutdown the source thread gracefully. Therefore I thought it should be safe to remove the test.